daml-foundations/daml-ghc/daml-stdlib-src/DA/Upgrade.daml

```diff
@@ -0,0 +1,87 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE OverlappingInstances #-}
+
+daml 1.2
+
+module DA.Upgrade
+  ( conv
+  , iso
+  )
+  where
+
+import DA.Generics
+
+-- | Convert data types that are isomorphic and have the same meta-data up to package id.
+conv : (Generic a repA, Generic b repB, Conv repA repB) => a -> b
+conv = to . cv . from
+
+-- | Generic representations that are isomorphic and have the same meta-data up to package id.
+class Conv a b where
+  cv : a x -> b x
```

Comment:
Should we call this `genericConv` or something similar?

Reply:
I'd keep it this way, since we're not exporting this class and if you're hacking on this module you know that this is a generic conversion.